src/java/org/apache/cassandra/transport/Message.java

```diff
@@ -225,12 +237,30 @@ protected boolean isTraceable()
         /**
          * Returns the time elapsed since this request was created. Note that this is the total lifetime of the request
          * in the system, so we expect increasing returned values across multiple calls to elapsedTimeSinceCreation.
+         * The elaspse time is calculated using the request time from the custom payload if it exists, otherwise it falls
+         * back to the time the request was created in the NTR stage (that is, when it was deserialized).
          *
          * @param timeUnit the time unit in which to return the elapsed time
          * @return the time elapsed since this request was created
          */
         protected long elapsedTimeSinceCreation(TimeUnit timeUnit)
```

Comment:
Need to fix queueTime/asyncQueue time as those are only valid against the post deserialization  `creationTimeNanos`

Reply:
I see, I was working on a way to maintain there original intent https://github.com/datastax/cassandra/pull/1672/commits/fe614ede7a68db7368510253c3b8f4b040928c44

but noted regarding the generic queue metrics. Will probably change there meaning like you said.